### PR TITLE
add dilation to avg_pool2d

### PR DESCRIPTION
**Pull request by rodfer0x80**

### CODE REVIEW

**Comment by geohot** [comment]
The whole point is to reenable the test!

**Comment by rodfer0x80** [comment]
Makes sense, will work on this

**Comment by rodfer0x80** [comment]
fix avg_pool2d passing onnx tests

**Comment by rodfer0x80** [comment]
-- add tensor avg pooling dilation feature
-- pass avg_pool2d and avg_pool3d onnx tests
-- hotfix onnx padding

**Comment by rodfer0x80** [comment]
- Torch doesn't support dilation in avg_pool but does in max_pool:
    torch.nn.AvgPool2d(kernel_size, stride=None, padding=0, ceil_mode=False, count_include_pad=True, divisor_override=None)
- Do the ONNX test suffice or do we need avg_pool tests for dilation in test/test_ops.py?
